fg21sim/extragalactic/pointsource/base.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2016 Zhixian MA <zxma_sjtu@qq.com>
+# MIT license
+
+"""
+Simulation of point sources for 21cm signal detection
+
+Point sources types
+-----
+1. Star forming (SF) galaxies
+2. Star bursting (SB) galaxies
+3. Radio quiet AGN (RQ_AGN)
+4. Faranoff-Riley I (FRI)
+5. Faranoff-Riley II (FRII)
+
+References
+------------
+[1] Wilman R J, Miller L, Jarvis M J, et al.
+    A semi-empirical simulation of the extragalactic radio continuum sky for
+    next generation radio telescopes[J]. Monthly Notices of the Royal
+    Astronomical Society, 2008, 388(3):1335–1348.
+[2] Jelic et al.
+[3] Healpix, healpy.readthedocs.io
+"""
+# Import packages or modules
+import os
+# import healpy as hp  # healpy
+import numpy as np
+import time
+from pandas import DataFrame
+import astropy.units as au
+# Custom module
+import psparams
+
+# Defination of the base class
+class pointsource:
+    """
+    The basic class of point sources
+
+    Parameters
+    ----------
+    z: float
+        Redshift, z ~ U(0,20)
+    dA: au.Mpc
+        Angular diameter distance, which is calculated according to the cosmology
+        constants. In this work, it is calculated by module basic_params
+    theta: au.rad
+        The colatitude angle in the spherical coordinate system
+    phi: au.rad
+        The longtitude angle in the spherical coordinate system
+    area: au.sr
+        Area of the point sources, sr = rad^2
+
+    Functions
+    ---------
+    gen_sgl_ps
+        Generate single ps
+    save_as_csv
+    """
+    #Init
+    z = 0
+    dA = 0 * au.Mpc * 0
+    theta = au.deg * 0
+    phi = au.deg * 0
+    area = au.sr * 0
+    Columns = []
+    nCols = 0
```

Comment:
Move these initialization to ``__init__()``.

For float number, explicitly use ``0.0``.

Place the units *behind* the number.